custom/covid/management/commands/add_assignment_cases.py

```diff
@@ -0,0 +1,78 @@
+import uuid
+from xml.etree import cElementTree as ElementTree
+
+from custom.covid.management.commands.update_cases import CaseUpdateCommand
+
+from casexml.apps.case.mock import CaseBlock
+from dimagi.utils.chunked import chunked
+
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.apps.locations.models import SQLLocation
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+
+
+BATCH_SIZE = 100
+DEVICE_ID = __name__ + ".add_assignment_cases"
+
+
+def should_skip(case, location_obj, active_location):
+    if location_obj:
+        return case.closed or (location_obj.name != active_location)
+    else:
+        return True
+
+
+def needs_update(case):
+    return case.get_case_property('is_assigned_primary') != 'yes' or \
+        case.get_case_property('is_assigned_temp') != 'yes'
+
+
+def find_owner_id(case, accessor):
+    checkin_case = accessor.get_case(case.assigned_to_primary_checkin_case_id)
+    return checkin_case.get_case_property('hq_user_id')
+
+
+class Command(CaseUpdateCommand):
+    help = "Creates assignment cases for cases of a specified type in an active location"
+
+    def case_block(self, case, owner_id, assignment_type):
+        case_id = uuid.uuid4().hex
+        return ElementTree.tostring(CaseBlock.deprecated_init(
+            create=True,
+            case_id=case_id,
+            case_type='assignment',
+            owner_id=owner_id,
+            index={'parent': (case.get_case_property('case_type'), case.case_id, "extension")},
+            update={"assignment_type": assignment_type},
+        ).as_xml()).decode('utf-8')
+
+    def update_cases(self, domain, case_type, user_id, active_location):
+        case_ids = self.find_case_ids_by_type(domain, case_type)
```

Comment:
There should be a dbaccessor that lets you filter by owner at the db level, maybe https://github.com/dimagi/commcare-hq/blob/master/corehq/form_processor/interfaces/dbaccessors.py#L304, that would save the need to do all this filtering in python